### PR TITLE
Fix bugs, TYPO3 v11 support, add functional tests and test setup

### PR DESCRIPTION
**Pull request by sypets**
* TYPO3 v11 support
* Fix issue #2 
* Add functional tests and test setup

------

A typical setup using runTests.sh is used as explained in the official documentation: https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/Testing/ExtensionTesting.html (this is slightly outdated but is being updated as we speak).

with 

* a GitHub CI file in .github/workflows/ci.yml to run the tests automatically in the future for each PR
* .attributes file so that the Tests are not packaged when installing the extension
* .gitignore so that the build files (.Build, composer.lock etc.) are not included in this repo
* runTests.sh & friends in Build directory
* functional tests in Tests/Functional which check if the fix in PR #3 works and some more basic testing

To execute the functional tests locally:

```
Build/Scripts/runTests.sh -s composerInstallMax -p 8.1
Build/Scripts/runTests.sh -s functional -p 8.1
```

The build can be removed with
```
rm -rf .Build
rm composer.lock

```

To not bloat up this PR too much, extra checks were not added, but are recommended:

* setup php-cgl-fixer and .editorconfig for coding guidelines with [typo3/coding-standards](https://packagist.org/packages/typo3/coding-standards)
* unit test. For this, the xml and boostrap file must be added to Build/phpunit (from typo3/testing-framework)
* run `composer validate`
* run lint
* install `phpstan` and execute checks
* add CONTRIBTING.md with info how to execute the checks and tests locally

### CODE REVIEW

**Review by sbuerk — CHANGES_REQUESTED** [review]

File: Build/Scripts/runTests.sh [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+            - composerCoreVersion: "composer require --no-install typo3/minimal:"coreVersion"
+            - cgl: test and fix all core php files
+            - cglGit: test and fix latest committed patch for CGL compliance
+            - lint: PHP linting
+            - phpstan: phpstan tests
+            - phpstanGenerateBaseline: regenerate phpstan baseline, handy after phpstan updates
+            - unit (default): PHP unit tests
+            - functional: functional tests
+
+    -t <composer-core-version-constraint>
```
[the remark this answers is not in the document]
Comment (sypets) [reply Build/Scripts/runTests.sh]: -t is currently not used. I removed / commented out the commands which are not used or not fully supported right now. Configuration files for phpstan and php-cs-fixer are also not supplied yet. Can be done with followup patch.

File: Build/Scripts/cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (sypets) [reply Build/Scripts/cleanup.sh]: Added -s cleanBuild

File: Build/Scripts/cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+# convenience script for cleaning up after running test suite locally
+
+composer config --unset platform.php
+composer config --unset platform
+
+rm -rf .Build
```
[the remark this answers is not in the document]
Comment (sypets) [reply Build/Scripts/cleanup.sh]: Done.

> When removing composer.lock, removing the vendor folder would make sense I guess.

The vendor folder is in .Build and that is already removed.

File: Build/Scripts/runTests.sh [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+Options:
+    -s <...>
+        Specifies which test suite to run
+            - composerInstall: "composer install"
+            - composerInstallMax: "composer update", with no platform.php config.
+            - composerInstallMin: "composer update --prefer-lowest", with platform.php set to PHP version x.x.0.
+            - composerValidate: "composer validate"
+            - composerCoreVersion: "composer require --no-install typo3/minimal:"coreVersion"
+            - cgl: test and fix all core php files
+            - cglGit: test and fix latest committed patch for CGL compliance
```
[the remark this answers is not in the document]
Comment (sypets) [reply Build/Scripts/runTests.sh]: Done. Commented out.

File: Build/testing-docker/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+    volumes:
+      - ${ROOT_DIR}:${ROOT_DIR}
+    working_dir: ${ROOT_DIR}
+    command: >
+      /bin/sh -c "
+        if [ ${SCRIPT_VERBOSE} -eq 1 ]; then
+          set -x
+        fi
+        php -dxdebug.mode=off .Build/bin/php-cs-fixer fix -v ${CGLCHECK_DRY_RUN} --path-mode intersection \
+          --config=Build/.php-cs-fixer.php .
```
[the remark this answers is not in the document]
Comment (sypets) [reply Build/testing-docker/docker-compose.yml:219]: dito.

**Comment by sypets** [comment]
@sbuerk Thank you for review! Should all be fixed now.

**Comment by franzkugelmann** [comment]
Thanks @sypets and @sbuerk! Beware, i renamed the branch master to main. Looking forward to further collaboration on the unduplicator.

**Pull request merged**